File: api/autocomplete.js
export default async function handler(req, res) {
  res.setHeader('Access-Control-Allow-Origin', '*');
  res.setHeader('Access-Control-Allow-Methods', 'GET, OPTIONS');
  res.setHeader('Access-Control-Allow-Headers', 'Content-Type');

  if (req.method === 'OPTIONS') {
    return res.status(200).end();
  }

  if (req.method !== 'GET') {
    return res.status(405).json({ error: 'Method not allowed' });
  }

  const { prefix = '', limit = 8 } = req.query;

  if (!prefix.trim()) {
    return res.json({ query: prefix, items: [] });
  }

  // Return mock data - Redis is not connected
  const items = [
    { id: '1', title: `${prefix} Art`, name: `${prefix} Art`, category: 'Art' },
    { id: '2', title: `${prefix} Design`, name: `${prefix} Design`, category: 'Design' }
  ];

  res.json({ query: prefix, items });
}

File: BACKEND/internal/handlers/autocomplete.go
package handlers

import (
	"encoding/json"
	"net/http"
	"strconv"

	redisClient "github.com/siddheshRajendraNimbalkar/collage-prject-backend/internal/redis"
)

type AutocompleteResponse struct {
	Query string                   `json:"query"`
	Items []map[string]string      `json:"items"`
}

func AutocompleteHandler(w http.ResponseWriter, r *http.Request) {
	// Set CORS headers
	w.Header().Set("Access-Control-Allow-Origin", "*")
	w.Header().Set("Access-Control-Allow-Methods", "GET, OPTIONS")
	w.Header().Set("Access-Control-Allow-Headers", "Content-Type")
	
	if r.Method == http.MethodOptions {
		w.WriteHeader(http.StatusOK)
		return
	}
	
	if r.Method != http.MethodGet {
		http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
		return
	}

	prefix := r.URL.Query().Get("prefix")
	limitStr := r.URL.Query().Get("limit")
	offsetStr := r.URL.Query().Get("offset")
	
	limit := 10
	if limitStr != "" {
		if l, err := strconv.Atoi(limitStr); err == nil && l > 0 {
			limit = l
		}
	}
	
	offset := 0
	if offsetStr != "" {
		if o, err := strconv.Atoi(offsetStr); err == nil && o >= 0 {
			offset = o
		}
	}

	results, err := redisClient.AutocompleteSearchWithOffset(prefix, limit, offset)
	if err != nil {
		http.Error(w, "Search failed", http.StatusInternalServerError)
		return
	}

	response := AutocompleteResponse{
		Query: prefix,
		Items: results,
	}

	w.Header().Set("Content-Type", "application/json")
	w.Header().Set("Access-Control-Allow-Origin", "*")
	w.Header().Set("Access-Control-Allow-Methods", "GET, OPTIONS")
	w.Header().Set("Access-Control-Allow-Headers", "Content-Type")

	if err := json.NewEncoder(w).Encode(response); err != nil {
		http.Error(w, "Failed to encode response", http.StatusInternalServerError)
	}
}